src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.VariableIdentifier.cs

```diff
@@ -24,28 +25,58 @@ public VariableIdentifier(Symbol symbol, int containingSlot = 0)
 
             public bool Exists
             {
-                get { return (object)Symbol != null; }
+                get { Debug.Assert((object)Symbol != null);  return (object)Symbol != null; }
             }
+
             public override int GetHashCode()
             {
-                return Roslyn.Utilities.Hash.Combine(Symbol?.OriginalDefinition, ContainingSlot);
+                Debug.Assert((object)Symbol != null);
+
+                int currentKey = ContainingSlot;
+                int thisIndex = Symbol.MemberIndex;
+                return (thisIndex < 0) ?
+                    Hash.Combine(Symbol.OriginalDefinition, currentKey) :
+                    Hash.Combine(thisIndex, currentKey);
```

Comment:
>Hash.Combine(thisIndex, currentKey); [](start = 20, length = 36)

nit: consider factoring `return Hash.Combine(thisIndex < 0 ? Symbol.OriginalDefinition : thisIndex, currentKey);`

Reply:
Unfortunately, the `Hash.Combine` calls are to different overloads.

---
In reply to: [259053161](https://github.com/dotnet/roslyn/pull/33562#discussion_r259053161) [](ancestors = 259053161)